### PR TITLE
Build with Gradle 7.0.

### PR DESCRIPTION
**Pull request by autonomousapps**
Not to be merged until I'm ready to raise the minimum supported version to Gradle 7. That is unlikely to happen soon.

### CODE REVIEW

File: src/functionalTest/groovy/com/autonomousapps/AbstractFunctionalSpec.groovy [hunk 1]
```diff
@@ -59,15 +59,17 @@ abstract class AbstractFunctionalSpec extends Specification {
   }
 
   protected static boolean isCompatible(GradleVersion gradleVersion, AgpVersion agpVersion) {
+    // TODO can't find docs on min Gradle version for AGP 7
+
     if (agpVersion >= AgpVersion.version('7.0.0')) {
-      // TODO can't find docs on min Gradle version.
       return gradleVersion >= GradleVersion.version('6.8.0')
```
Comment (plastiv) [inline src/functionalTest/groovy/com/autonomousapps/AbstractFunctionalSpec.groovy:65]: nit: AGP 7.x is meant to work with Gradle 7.x APIs

https://android-developers.googleblog.com/2020/12/announcing-android-gradle-plugin.html

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/sdu5eubd5bhbi

File: src/functionalTest/groovy/com/autonomousapps/AbstractFunctionalSpec.groovy (hunk 1 above)
Comment (autonomousapps) [reply src/functionalTest/groovy/com/autonomousapps/AbstractFunctionalSpec.groovy:65]: Sure, but in my initial testing it's not an auto fail with 6.8, which is interesting.

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/bwdhem5rvouxe

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/2oixfh5ev6iqo

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/pxva4frbpsndw

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/yhyup2peklzvg

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/i5h2yz65odqag

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/mtsrtstr6zcp6

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/sijxng4pyjxum

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/wdtabiftttvk4

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/ey3qkzisko66q

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/kprnbtwfnmxc2

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/raec6dnkbmej6

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/sfcwpo4y6evwu

**Comment by autonomousapps** [comment]
Build failed https://gradle.com/s/ffddfdzsgywpm

**Pull request merged**